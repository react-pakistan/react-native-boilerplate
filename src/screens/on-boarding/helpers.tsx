/* eslint-disable global-require */

import React from 'react';
import { Image } from 'react-native';
import { Page } from 'react-native-onboarding-swiper';

export const onBoardingData : Array<Page> = [
  {
    backgroundColor: '#00FFF2',
    image: <Image source={require('../../assets/images/illustration.png')} />,
    subtitle: `A generic, Commons library for React Native apps,
      following modern approach by building your project's
      UI in isolation using state of the art Storybook along
      with React Native.`.replace(/[\n\r]+ */g, ' '),
    subTitleStyles: {
      color: 'black',
    },
    title: 'React Native Commons Collection',
    titleStyles: {
      color: 'black',
    },
  },
  {
    backgroundColor: '#FF0074',
    image: <Image source={require('../../assets/images/illustration.png')} />,
    subtitle: `A generic to specific, UI/UX library for React Native apps,
      following modern approach by building your project's
      UI in isolation using state of the art Storybook along
      with React Native.`.replace(/[\n\r]+ */g, ' '),
    subTitleStyles: {
      color: 'black',
    },
    title: 'React Native UI Collection',
    titleStyles: {
      color: 'black',
    },
  },
  {
    backgroundColor: '#FFC200',
    image: <Image source={require('../../assets/images/illustration.png')} />,
    subtitle: `A generic, Icons library for React Native apps,
      following modern approach by building your project's
      UI in isolation using state of the art Storybook along
      with React Native.`.replace(/[\n\r]+ */g, ' '),
    subTitleStyles: {
      color: 'black',
    },
    title: 'React Native Icon Collection',
    titleStyles: {
      color: 'black',
    },
  },
  {
    backgroundColor: '#0DFF00',
    image: <Image source={require('../../assets/images/illustration.png')} />,
    subtitle: `A generic, Logo library for React Native apps,
      following modern approach by building your project's
      UI in isolation using state of the art Storybook along
      with React Native.`.replace(/[\n\r]+ */g, ' '),
    subTitleStyles: {
      color: 'black',
    },
    title: 'React Native Logo Collection',
    titleStyles: {
      color: 'black',
    },
  },
  {
    backgroundColor: '#0DFF00',
    image: <Image source={require('../../assets/images/illustration.png')} />,
    subtitle: `A generic, Emoji library for React Native apps,
      following modern approach by building your project's
      UI in isolation using state of the art Storybook along
      with React Native.`.replace(/[\n\r]+ */g, ' '),
    subTitleStyles: {
      color: 'black',
    },
    title: 'React Native Emoji Collection',
    titleStyles: {
      color: 'black',
    },
  },
];
