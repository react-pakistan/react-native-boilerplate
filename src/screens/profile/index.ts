export * from './profile';
export * from './type';
