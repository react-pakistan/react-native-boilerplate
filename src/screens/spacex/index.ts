export * from './spacex';
export * from './type';
