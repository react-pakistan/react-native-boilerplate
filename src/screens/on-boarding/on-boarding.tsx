import React, { Dispatch, ReactElement } from 'react';
import Onboarding from 'react-native-onboarding-swiper';
import { useDispatch } from 'react-redux';
import { showOnboarding } from '../../redux/slices/main-slice';
import { Next, Skip } from './components';
import { onBoardingData } from './helpers';

export const OnBoarding = () : ReactElement => {
  // dispatch
  const dispatch = useDispatch();

  return (
    <Onboarding
      SkipButtonComponent={Skip}
      NextButtonComponent={Next}
      onDone={() => dispatch(showOnboarding())}
      onSkip={() => dispatch(showOnboarding())}
      pages={onBoardingData}
    />
  );
};
