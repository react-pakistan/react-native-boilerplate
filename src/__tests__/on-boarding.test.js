import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { OnBoarding } from '../screens/on-boarding';

const fakeInitialState = {
  showAppReducer: {
    showApp: true,
  },
};

const mockStore = configureStore();

test('renders Dashboard component correctly', () => {
  const renderer = new ShallowRenderer();
  const store = mockStore(fakeInitialState);
  renderer.render(
    <Provider store={store}>
      <OnBoarding />
    </Provider>,
  );
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
