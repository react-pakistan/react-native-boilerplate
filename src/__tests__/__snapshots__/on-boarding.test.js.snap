// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard component correctly 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <OnBoarding />
</Context.Provider>
`;
